### ISSUE TITLE
PR not syncing with JIRA

### ISSUE THREAD

**Issue by mccannical** [atlassian/github-for-jira#674]
We have been having this issue on and off now for at least a week. 

Is there a limitation on how often the sync should occur? 

The status page for our app shows that the last sync was 3 days ago.  Does that indicate that PRs should be linked up to 3 days ago, or is this just the last time a full sync was done?

**Comment by mboudreau** [comment]
The "sync" in it's current form is meant as an indicator of the full historical data sync.  From that point on, it would be using webhooks to get any new current data.  We're also in the process (now that we have ownership) of redoing the syncing/webhook mechanism to make it much more accurate, stable and resilient

What's your jira domain?  We can investigate in our logs to see if any issues arises with yours and try to address it ASAP.  Thanks.

**Comment by mccannical** [comment]
confluentinc

**Comment by juliebartlett** [comment]
We're having this issue at doxo as well.

**Comment by thombergs** [comment]
same issue as in #673

**Comment by martynamaron** [comment]
I'm seeing the same flakiness - sometimes the `branch` is linked, sometimes just the merge commit, but never the PR and the ticket status never changes :(

**Comment by stuartpickering** [comment]
We are seeing flakiness as well. jira domain: thebind
This flakiness seems to have started within the last month or so. Have tried a full sync to fix the issue as well.
We are merging pull requests that are tied to tickets, and they still show as 'Open' in the tickets, days later.

**Comment by celinakRS** [comment]
Since updating to the latest version to fix our PR status messages if a Comment was left, we've seen quite a few situations where the status of the merge is no longer accurate.  In one example:

- The PR was created with comment (JIRA shows as OPEN) -- correct
- PR was self assigned
- PR review was required from user
- user approved
- PR was merged (JIRA shows as OPEN) -- incorrect
- feature branch deleted

5 days later since the merge and this PR still shows as OPEN in JIRA
Plus, the ticket also has a 2nd PR that is 'MERGED' but the ticket only shows the one PR .
![image](https://user-images.githubusercontent.com/83659124/131722121-8f858376-75eb-4bdc-9c3c-bfcc0058097f.png)

jira domain: retailsuccess

**Comment by thombergs** [comment]
@celinakRS @stuartpickering thanks for providing your information about the issues. We're looking into it.

**Comment by thombergs** [comment]
@celinakRS @stuartpickering @martynamaron @juliebartlett @mccannical we have added some diagnostics to the code, but we're still researching what could be the cause of this.

Hey Roy Ngan

I'm from the team investigating this issue. It has also been reported by some other users that pull requests don't arrive in Jira, but we're still trying to isolate the cause for this. Could you please let us know if a PR is failing to reach Jira again and provide the URL and title of the PR? This would help us identify the cause with our newly added diagnostics.

If you're not comfortable sharing the URL and title of the PR here, feel free to share it via a support request at https://support.atlassian.com and refer to this issue here.

Sorry about this and thanks you for your help!

**Comment by adrianne-helpscout** [comment]
@thombergs I first raised this issue some days ago in issue #638 per your suggestion I opened a support request. The support agent let me know that the issue has been escalated and is now being tracked by [JSWCLOUD-21862](https://jira.atlassian.com/browse/JSWCLOUD-21862)

**Comment by thombergs** [comment]
@adrianne-helpscout thanks! Post them here on in the Jira issue. In any case, please try to mention me so I get the update :)

**Comment by thombergs** [comment]
Hey all,

we have identified and just fixed an issue with PRs not reaching Jira. Please let us know if you still find any issues with this (or to report that it's working now :)).

**Background:**
During PR processing, the app extracts Jira issue keys from the PR description (to expand them into hyperlinks) and then checks against the Jira API if those issues exist. If **just one**  of the issues did not exist, the processing would abort, resulting in the PR (or PR status update) not reaching Jira. This bug only became visible after we changed the issue key parser to find more issue key patterns. The new issue key parser would find the issue key `ab-139` in the UUID `123e45ab-139b-12d3-a456-426614174000`, for example. And all PRs that contain images contain UUIDs, because GitHub creates URLs with UUIDs for all uploaded images. So all PRs containing images would be affected by this.

**Comment by mboudreau** [comment]
Thank you all for your patience as we tried to pinpoint the cause of this issue. If any historical data is off, please do a full resync under the Jira managed apps section for our app and it should fix itself shortly.

As some of you may know, we've recently done an ownership migration from GitHub to Atlassian and these issues have cropped up as part of the migration. We're very sorry that it did and have inconvenienced you, but it was a necessary step for us to have more control and be able to deliver better features faster.

This was a bumpy start to be sure, but we're hoping that we can instill some confidence in us moving forward with what we have planned for the app.  Please stay tuned and again, thank you for your patience.

Cheers!

**Comment by celinakRS** [comment]
> Hey all,
> 
> we have identified and just fixed an issue with PRs not reaching Jira. Please let us know if you still find any issues with this (or to report that it's working now :)). Sorry for this issue and thanks for your patience.
> 

Is there a new app update we need to install for this or was the fix on your backend?

**Comment by adrianne-helpscout** [comment]
@thombergs Thanks for the information. I will run a resync and will let you know if we see any new issues next week that continue to exhibit this behavior.

**Comment by thombergs** [comment]
> Is there a new app update we need to install for this or was the fix on your backend?
@celinakRS it was a backend fix, nothing to do on your end, except maybe to trigger a sync to catch the missed PRs.

**Comment by eallanfb** [comment]
> ease let us know if you still find any issues with t

Hello Tom,

Good day, it seems that my Jira Github integration still not working on my end. I dont know if it is because of my 625 reports that is not yet sync.

**Comment by thombergs** [comment]
@eallanfb The fix will only apply to new PRs that are being created in GitHub. If you want to backfill the PRs from last week that were affected by the bug, you will need to trigger a "sync" in the configuration of the GitHub integration. In the "Retry" column, select "Full" and click on the "Submit" button. 

Please let us know if you are still experiencing issues with new PRs that don't show up in Jira.

**Comment by eallanfb** [comment]
Hello Tom,

As per your instruction to do the Github configuration to  trigger a "sync" in the configuration of the GitHub integration. In the "Retry" column, select "Full" and click on the "Submit" button.

However, it did failed and what I am trying to do again is resync for another and see how it goes.

Hope this case will remain open until everything is resolve.

Thank you
Allan

**Comment by eallanfb** [comment]
Hello Tom,

It happened again after reaching repos sync 99/625 the resync failed and I have tried this twice and error still the same. Hope you have resolution on this issue.

Thank you,
Allan

**Comment by eallanfb** [comment]
Hello Tom,

I tried to do as you instructed. However, after reaching the resync of
99/625 respo sync, it shows FAILED and I have tried this twice and getting
the same error.

Also, as you have said that the Jira-Github Integration bug is now
resolved, do you mean that the we can now see the name corresponding card
in Jira and the branch will show up there?

Sorry, I am very new to this Jira and that is why I am asking too
many questions.

Thank you,
Allan

**Comment by mboudreau** [comment]
@eallanfb Hm, interesting.  We'll look into it, what's your jira domain so we can look at the logs?

**Comment by eallanfb** [comment]
theladders.atlassian.net

**Comment by mboudreau** [comment]
Yeah, I can see the errors, will need to investigate this further.  Thanks!

**Comment by eallanfb** [comment]
Hello Michel,

Thank you and please update me and if you could provide me a possible
timeframe on the resolution of the issue.

Also, as you have said that the Jira-Github Integration bug is now
resolved, do you mean that we can now see the name corresponding card in
Jira and the branch will show up there?

Sorry, I am very new to this Jira and that is why I am asking too
many questions.

**Comment by mboudreau** [comment]
This seems to be a different bug.  We just need to identify what piece of the code is causing it, fix it, and then you should be good to do the sync.

Nah mate, not too many questions.  You're helping us fix some issues after our migration :)

**Comment by eallanfb** [comment]
Hello Michel,

Do we have a timeframe on when the issue will be resolved?

Also, if you could answer my questions about the previous jira and github
migration bug.

As you have said that the Jira-Github Integration bug is now resolved, do
you mean that we can now see the name corresponding card in Jira and the
branch will show up there?

Hope to hear from you soon

Thank you and kind regards
Allan

**Comment by mboudreau** [comment]
@eallanfb We've identified the issue and fixed it in the code.  We're currently testing it and hope to deploy it soon.

**Comment by eallanfb** [comment]
Hello Michel,

Thank you and I'm looking forward to getting the issue fixed.

Please let us know once the fix is ready or deployed and we can try it on
our end.

Thank you,
Allan

**Comment by eallanfb** [comment]
Hello Michel,

Good day to you and I hope everything is well with you. I would like to
ask, do we have any update on this?

Hope to hear from you soon

Thank you and kind regards
Allan

**Comment by juliebartlett** [comment]
It seems like this gets fixed, and then it breaks again.

**Comment by eallanfb** [comment]
> It seems like this gets fixed, and then it breaks again.

Does this mean, that there issue again? and its not really fix it yet?

**Comment by eallanfb** [comment]
> It seems like this gets fixed, and then it breaks again.

Also, it affects our automatically creating JIRA issues when using the @help label

any resolution on this as well?

**Comment by celinakRS** [comment]
Back to the comment I posted 2 weeks ago : https://github.com/atlassian/github-for-jira/issues/674#issuecomment-910533231

I've done two full sync's since then (one was just 3 days ago) and this is still occurring.  

Example: 
PR created 6 days ago --> status in github - merged --> status in jira - merged
PR created 1 day ago --> status in github - merged --> status in jira - open 

It appears the 2nd PR status is not re-sync'd.

**Comment by celinakRS** [comment]
And another example:

PR created 11 days ago --> status in github - merged --> status in jira - open

PR history:
review requested for feature branch to develop  - 11 days ago
catch up merge from develop into this feature branch 5 days ago
PR approved
feature branch merged into develop

**Comment by mboudreau** [comment]
Sorry folks, we've hit some snags in the queuing/worker processing mechanism combined with Github's API rate limiting which made the syncs unstable in the last couple of weeks.  We have been working tirelessly to get the stability that you in fact deserve. 

We've identified several major issues with the queue processing code/infrastructure that we're still trying to improve in the coming weeks, but we believe we have addressed the biggest blockers for the time being.  If data is missing, please try to do a "full" resync.  If you're still seeing issues, please let us know and we'll try to address it individually.

Thank you again for your patience and I want to again say how sorry we are to have caused these headaches for you.

**Comment by eallanfb** [comment]
Hello  Michel,

I am running a resync again now and let see if this will continue and if
not, I will get back to you again.

Thank you,
Allan

**Comment by eallanfb** [comment]
Hello Michel,

Yes I did a resync but not like the previous resync, it reached 99/625 but
now, it only reaches 42 then failed again.

I would like to ask when this will be resolved or if there is any
resolution being done for this issue?

Hope to hear from you soon

Thank you and best regards
Allan

**Comment by eallanfb** [comment]
Do we have any update on this?

**Comment by eallanfb** [comment]
Hello Michel,

Any update on this, this has been on going and no resolution provided. This affect our operations and this is not good support for Jira team.

**Comment by rachellerathbone** [comment]
Hey @eallanfb. We recently deployed some changes that appear to have fixed all failing syncs. Could you please trying syncing again and let me know if you continue experiencing issues?

**Comment by eallanfb** [comment]
Hello Rachelle, Everything seems to be working fine now after resync of 625 and there was no error. Thank you for this.

**Comment by rachellerathbone** [comment]
That is great news @eallanfb and apologize for any inconvenience that was caused. 

@mccannical, @juliebartlett, @martynamaron, @stuartpickering, @celinakRS, and @adrianne-helpscout please let me know if you are facing issues. We believe we have resolved the causes behind these issues but want to make sure the app is working as expected for you all so we can provided further assistance if needed.

**Comment by celinakRS** [comment]
@rachellerathbone - So far so good, the only other remaining issue is now a feature request on the roadmap.  Thank you for your help!!

- https://github.com/atlassian/github-for-jira/issues/765
